### PR TITLE
URL bar autocomplete mouseover does not interfere with typed URL (fixes #3012)

### LINKED ISSUE #3012: Autofill should not happen on URL bar unless item in autocomplete list is clicked
From support:
> When I try to paste a hyperlink or type something in I need to ensure the cursor is far away from the drop-down history in the URL address bar, otherwise it auto-fills with one of those links and then starts appending what I'm typing onto it. The auto-fill shouldn't happen unless I click on the hyperlink from the history when the drop down box appears.

Sometimes I notice this too.

### PR DESCRIPTION
**Pull request by willy-b**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

Fixes #3012: Autofill should not happen on URL bar unless item in autocomplete list is clicked

### CODE REVIEW

File: js/actions/windowActions.js [hunk 1]
```diff
@@ -489,8 +489,8 @@ const windowActions = {
   setUrlBarSuggestions: function (suggestionList, selectedIndex) {
     dispatch({
       actionType: WindowConstants.WINDOW_SET_URL_BAR_SUGGESTIONS,
-      suggestionList,
-      selectedIndex
+      suggestionList: suggestionList,
+      selectedIndex: selectedIndex
```
Comment (bbondy) [inline js/actions/windowActions.js]: this change doesn't seem necessary, any reason for it?

File: js/components/urlBar.js [hunk 2]
```diff
@@ -250,9 +256,13 @@ class UrlBar extends ImmutableComponent {
...
   onActiveFrameStop () {
-    this.restore()
-    windowActions.setUrlBarSelected(true)
-    windowActions.setUrlBarActive(false)
+    if (this.isFocused()) {
+      windowActions.setUrlBarActive(false)
+      if (!this.shouldRenderUrlBarSuggestions) {
+        this.restore()
+        windowActions.setUrlBarSelected(true)
+      }
```
Comment (bbondy) [inline js/components/urlBar.js]: I think there's some kind of a rebase issue here, please rebase from current master

**Comment by bbondy** [comment]
would you mind adding a test for this as well? Thanks.
It could go in here: `test/components/navigationBarTest.js`
and you could use moveToObject for a the mouse move on top of the suggestions

**Comment by willy-b** [comment]
Thanks @bbondy for all the feedback! 
Closing this in favor of a new PR #3225 which is rebased against master and has the changes you requested and a test.

**Pull request closed without merging**